skiko/src/nativeMain/kotlin/org/jetbrains/skia/impl/Native.native.kt

```diff
@@ -284,3 +281,10 @@ private external fun initCallbacks(
     callVoid: COpaquePointer,
     dispose: COpaquePointer
 )
+
+private fun convertToZeroTerminatedString(string: String): ByteArray {
+    // encodeToByteArray encodes to utf8
+    val utf8 = string.encodeToByteArray()
+    // TODO Remove array copy, use `skString(data, length)` instead of `skString(data)`
```

Comment:
Is there still no way to avoid double-copy in K/N?

Reply:
We can do it, with this TODO comment. But for now it is safe to stay double copy